### ISSUE TITLE
Run failure caused by JDK version

### ISSUE THREAD

**Issue by liluyue** [kotest/kotest#4089]
**version of Kotest are:5.9.1**
```
com/example/result/kotlin/base/TextTest has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 52.0

```

**Comment by Kantis** [comment]
Looks like you're compiling with jdk17 but running tests with jdk8

**Comment by liluyue** [comment]
> Looks like you're compiling with jdk17 but running tests with jdk8

yes.My current environment is jdk17, I don't know where this jdk8 comes from

**Comment by Kantis** [comment]
Can you check your project settings, is it configured for jdk17?

<img width="909" alt="image" src="https://github.com/kotest/kotest/assets/316857/98010390-5905-4d63-9b55-c440f04761c2">

**Comment by liluyue** [comment]
<img width="941" alt="image" src="https://github.com/kotest/kotest/assets/19216288/17d8732e-44ba-4c61-bd26-c88ffe5087ba">
<img width="976" alt="image" src="https://github.com/kotest/kotest/assets/19216288/37a0420d-3c0c-44cd-a22f-94c09a8a65f6">
<img width="1059" alt="image" src="https://github.com/kotest/kotest/assets/19216288/c6e253da-dbb3-4e56-b22f-e0e5ebd14938">

**Comment by Kantis** [comment]
And your build file?
How do you start your tests?

**Comment by liluyue** [comment]
click button
<img width="1244" alt="image" src="https://github.com/kotest/kotest/assets/19216288/0f0dea6d-2319-4426-adff-bdaa1d3414e7">

**Comment by Kantis** [comment]
If you edit that run configuration, you should see the java version used to run the tests

**Comment by liluyue** [comment]
<img width="1069" alt="image" src="https://github.com/kotest/kotest/assets/19216288/6e1efa90-6927-4b7b-a09f-f42c50655422">
<img width="1309" alt="image" src="https://github.com/kotest/kotest/assets/19216288/61d278f5-3965-4e72-9853-eb1ff37a4c9a">
After making the changes, the errors have changed

**Comment by liluyue** [comment]
There is no problem using the gradle command. It should be [plugin](https://plugins.jetbrains.com/plugin/14080-kotest)The issue of.

**Comment by sksamuel** [comment]
> <img alt="image" width="1069" src="https://private-user-images.githubusercontent.com/19216288/338885031-6e1efa90-6927-4b7b-a09f-f42c50655422.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwMDcxMjksIm5iZiI6MTcyMzAwNjgyOSwicGF0aCI6Ii8xOTIxNjI4OC8zMzg4ODUwMzEtNmUxZWZhOTAtNjkyNy00YjdiLWEwOWYtZjQyYzUwNjU1NDIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDA1MDAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzN2FiMDAxYjkzZTFiMDMxOTgzNTU4NmVlYzlkNDhiMGU2ZGM4NWFjOTc0Mzg5NTE1ZTY1MjljZDQ3N2Q1NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.A2DDs9cmm7St9B_PQ-wgekF7-rzTx4Zx-SGOihzwRmQ"> <img alt="image" width="1309" src="https://private-user-images.githubusercontent.com/19216288/338885172-61d278f5-3965-4e72-9853-eb1ff37a4c9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwMDcxMjksIm5iZiI6MTcyMzAwNjgyOSwicGF0aCI6Ii8xOTIxNjI4OC8zMzg4ODUxNzItNjFkMjc4ZjUtMzk2NS00ZTcyLTk4NTMtZWIxZmYzN2E0YzlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDA1MDAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjMTBhM2QxMjc3MDk4NzkyZjlmMDNlZTFkYzAyM2I2NmU1OTg0NDFjNDZkNTMzNGRhNDY0YTYxZWI2NDBiYTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZdywCDbw4O29PAy3W1S-A_-0DWCo14YOMIGM6yQxPDQ"> After making the changes, the errors have changed

I am seeing this too sometimes. It is a config issue somewhere.

**Comment by edgarmiro** [comment]
I'm experiencing the same error. However, this situation only happens when I run a single test. When I run the whole test class it's working perfectly. 
The error is:
```
BlaBlaTest has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Then, if I change the run configuration to use JDK17 (The same version I have in my build) and run a single test:
```
java.lang.NoClassDefFoundError: kotlin/reflect/full/KClasses
	at io.kotest.mpp.JvmReflection.annotations(reflection.kt:35)
	at io.kotest.mpp.JvmReflection.annotations(reflection.kt:26)
	at io.kotest.engine.concurrency.ConcurrencyKt.isIsolate(concurrency.kt:22)
	at io.kotest.engine.ConcurrentTestSuiteScheduler.schedule(ConcurrentTestSuiteScheduler.kt:41)
	at io.kotest.engine.TestEngine$execute$innerExecute$1.invokeSuspend(TestEngine.kt:65)
	at io.kotest.engine.TestEngine$execute$innerExecute$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$innerExecute$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.TestEngineInitializedInterceptor.intercept(TestEngineInitializedInterceptor.kt:16)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.DumpConfigInterceptor.intercept(DumpConfigInterceptor.kt:21)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.WriteFailuresInterceptor.intercept(WriteFailuresInterceptor.kt:34)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.EmptyTestSuiteInterceptor.intercept(EmptyTestSuiteInterceptor.kt:27)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.ProjectTimeoutEngineInterceptor.intercept(ProjectTimeoutEngineInterceptor.kt:15)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.ProjectListenerEngineInterceptor.intercept(ProjectListenerEngineInterceptor.kt:23)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.ProjectExtensionEngineInterceptor$intercept$initial$1.invokeSuspend(ProjectExtensionEngineInterceptor.kt:22)
	at io.kotest.engine.interceptors.ProjectExtensionEngineInterceptor$intercept$initial$1.invoke(ProjectExtensionEngineInterceptor.kt)
	at io.kotest.engine.interceptors.ProjectExtensionEngineInterceptor$intercept$initial$1.invoke(ProjectExtensionEngineInterceptor.kt)
	at io.kotest.engine.interceptors.ProjectExtensionEngineInterceptor.intercept(ProjectExtensionEngineInterceptor.kt:34)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.SpecSortEngineInterceptor.intercept(SpecSortEngineInterceptor.kt:29)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.TestDslStateInterceptor.intercept(TestDslStateInterceptor.kt:15)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.interceptors.TestEngineStartedFinishedInterceptor.intercept(TestEngineStartedFinishedInterceptor.kt:22)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invokeSuspend(TestEngine.kt:71)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine$execute$execute$1$1.invoke(TestEngine.kt)
	at io.kotest.engine.TestEngine.execute$kotest_framework_engine(TestEngine.kt:77)
	at io.kotest.engine.TestEngineLauncher.async(TestEngineLauncher.kt:244)
	at io.kotest.engine.launcher.MainKt$main$1.invokeSuspend(main.kt:36)
	at io.kotest.engine.launcher.MainKt$main$1.invoke(main.kt)
	at io.kotest.engine.launcher.MainKt$main$1.invoke(main.kt)
	at io.kotest.common.RunBlockingKt$runBlocking$1.invokeSuspend(runBlocking.kt:3)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:277)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:95)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.common.RunBlockingKt.runBlocking(runBlocking.kt:3)
	at io.kotest.engine.launcher.MainKt.main(main.kt:34)
Caused by: java.lang.ClassNotFoundException: kotlin.reflect.full.KClasses
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 66 more

kotlin/reflect/full/KClasses
```

**Comment by erikhuizinga** [comment]
I face this same (or a very similar) issue.

### TL;DR

Add the following dependencies to fix the problem, at least in my case (described below).

```kotlin
testImplementation("org.opentest4j:opentest4j:$opentest4jVersion")
testImplementation("org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion")
```

---

### Longer investigation and reproduction of the problem.

I have the following setup:

- I'm in an Android library project
- I use Android Gradle Plugin 8.5.2
- I use JUnit 4.13.2
- I depend on `kotest-bom:5.9.1` and `io.kotest:kotest-runner-junit4` + property + data + assertions
- I have _not_ configured `useJUnitPlatform()`
- I have _not_ configured `useJUnit()`
- I have the Kotest IDE plugin installed, version 1.3.90-2024.2.2
- I run Android Studio 2024.2.2 stable
- I depend on Kotlin 2.0.10
- I use Gradle JVM toolchain 17 and the foojay toolchain resolver.

I have the following test class. Two out of the four tests fail deliberately, to check that it all runs well:

```kotlin
import io.kotest.matchers.shouldBe
import io.kotest.property.checkAll
import io.kotest.runner.junit4.FunSpec
import kotlin.test.assertEquals

class KotestTest : FunSpec({
    test("Succeeding test") {
        val result = 1 + 1
        assertEquals(2, result)
    }
    test("Failing test") {
        val result = 1 + 1
        assertEquals(3, result)
    }
    test("Succeeding kotest test") {
        checkAll<String> { str ->
            str shouldBe str
        }
    }
    test("Failing kotest test") {
        checkAll<String> { str ->
            str shouldBe "foo"
        }
    }
})
```

When I run Gradle task `testDebugUnitTest`, everything works fine: the tests run and the 2 assertions that we expect to fail do so.

When I run the test class directly, e.g. by clicking on it and pressing <kbd>⌃</kbd><kbd>⇧</kbd><kbd>R</kbd> (control + shift + R on macOS), I'm presented with the following 2 options:

<img width="380" alt="Image" src="https://github.com/user-attachments/assets/e0ebeb95-4647-4aca-afb5-76056a31764b" />

Note the icons: the first seems to be Gradle, the second Kotest. Choosing Gradle runs the test class just fine! 👍 Choosing Kotest (you may have to delete the Gradle run configuration first to see the option) fails with the already mentioned error: `java.lang.NoClassDefFoundError: kotlin/reflect/full/KClasses`.

So if the class isn't found, then maybe it should be added as a `testImplementation` dependency. I now add the `org.jetbrains.kotlin:kotlin-reflect` dependency and the Kotest run configuration to run only the class again. Now I get:

<img width="1065" alt="Image" src="https://github.com/user-attachments/assets/c931dc58-b3db-487d-8124-4effc9e52582" />

<details>
<summary>👈 Expand to see output / stacktrace / logs</summary>

```
Process finished with exit code 255

org.opentest4j.TestAbortedException
java.lang.ClassNotFoundException: org.opentest4j.TestAbortedException
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at io.kotest.engine.interceptors.MarkAbortedExceptionsAsSkippedTestInterceptor.intercept(MarkAbortedExceptionsAsSkippedTestInterceptor.kt:24)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.ExpectExceptionTestInterceptor.intercept(ExpectExceptionTestInterceptor.kt:18)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.TimeoutInterceptor.intercept(TimeoutInterceptor.kt:33)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.BlockedThreadTimeoutInterceptor.intercept(BlockedThreadTimeoutInterceptor.kt:79)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.CoroutineLoggingInterceptor.intercept(CoroutineLoggingInterceptor.kt:30)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.SoftAssertInterceptor.intercept(SoftAssertInterceptor.kt:27)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.AssertionModeInterceptor.intercept(AssertionModeInterceptor.kt:25)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.LifecycleInterceptor.intercept(LifecycleInterceptor.kt:50)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.TestCaseExtensionInterceptor$intercept$2.invokeSuspend(TestCaseExtensionInterceptor.kt:24)
	at io.kotest.engine.test.interceptors.TestCaseExtensionInterceptor$intercept$2.invoke(TestCaseExtensionInterceptor.kt)
	at io.kotest.engine.test.interceptors.TestCaseExtensionInterceptor$intercept$2.invoke(TestCaseExtensionInterceptor.kt)
	at io.kotest.engine.test.TestExtensions.intercept(TestExtensions.kt:148)
	at io.kotest.engine.test.interceptors.TestCaseExtensionInterceptor.intercept(TestCaseExtensionInterceptor.kt:24)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.BeforeSpecListenerInterceptor$intercept$runTest$1$2.invokeSuspend(BeforeSpecListenerInterceptor.kt:43)
	at io.kotest.engine.test.interceptors.BeforeSpecListenerInterceptor$intercept$runTest$1$2.invoke(BeforeSpecListenerInterceptor.kt)
	at io.kotest.engine.test.interceptors.BeforeSpecListenerInterceptor$intercept$runTest$1$2.invoke(BeforeSpecListenerInterceptor.kt)
	at io.kotest.engine.test.interceptors.BeforeSpecListenerInterceptor.intercept(BeforeSpecListenerInterceptor.kt:60)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.TestEnabledCheckInterceptor.intercept(TestEnabledCheckInterceptor.kt:31)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.CoroutineErrorCollectorInterceptor$intercept$3.invokeSuspend(CoroutineErrorCollectorInterceptor.kt:34)
	at io.kotest.engine.test.interceptors.CoroutineErrorCollectorInterceptor$intercept$3.invoke(CoroutineErrorCollectorInterceptor.kt)
	at io.kotest.engine.test.interceptors.CoroutineErrorCollectorInterceptor$intercept$3.invoke(CoroutineErrorCollectorInterceptor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at io.kotest.engine.test.interceptors.CoroutineErrorCollectorInterceptor.intercept(CoroutineErrorCollectorInterceptor.kt:33)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invokeSuspend(TestCaseExecutor.kt:100)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.TestCaseExecutor$execute$3$1.invoke(TestCaseExecutor.kt)
	at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invokeSuspend(coroutineDispatcherFactoryInterceptor.kt:57)
	at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invoke(coroutineDispatcherFactoryInterceptor.kt)
	at io.kotest.engine.test.interceptors.CoroutineDispatcherFactoryInterceptor$intercept$4.invoke(coroutineDispatcherFactoryInterceptor.kt)
	at io.kotest.engine.concurrency.FixedThreadCoroutineDispatcherFactory$withDispatcher$4.invokeSuspend(FixedThreadCoroutineDispatcherFactory.kt:59)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

</details>

So: one test does run, but the next aborts? 😕 👎 

Again, I proceed to add `org.opentest4j:opentest4j`, currently 1.3.0 is the latest version, as a `testImplementation` dependency. And now it works! ✨👍 

Then I try running the first test only, from the gutter or again by selecting it and pressing the keyboard shortcut and choose to run with Kotest. And that works:

<img width="1063" alt="Image" src="https://github.com/user-attachments/assets/35bef891-e504-4d43-be2a-51da378cf8bd" />

Note that the other tests do show up as ignored/disabled in the left test run window pane.

And the other tests also run fine on their own (failing assertions where expected, of course).

This probably means that the kotest junit4 runner (and possibly 5 too?) need to expose kotlin-reflect and opentest4j as `api` dependencies. And there may be other dependencies that I currently didn't need in my very simple test examples that may need to be included.

**Comment by sksamuel** [comment]
In 6.1, execute has moved to gradle runners. So if you update your project to 6.1 and use the latest intellij kotest plugin, this issue will go away.
Please reopen if you see otherwise.
Thanks for bug report.
